### Importando Librerias Necesarias

In [15]:
import pandas as pd
import numpy as np

### leyendo del dataset las hojas HECHOS y VICTIMAS

In [16]:
#tuve que instalar openpyxl para leer los archivos
hechos = pd.read_excel('../Datasets/homicidios.xlsx', sheet_name='HECHOS')
victimas =  pd.read_excel('../Datasets/homicidios.xlsx', sheet_name='VICTIMAS') 

In [17]:
hechos.head(5)

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA HH  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00  4   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00  1   
2  2016-0003           1 2016-01-03  2016   1   3  07:00:00  7   
3  2016-0004           1 2016-01-10  2016   1  10  00:00:00  0   
4  2016-0005           1 2016-01-21  2016   1  21  05:20:00  5   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE  ...  Altura  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...     NaN   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...     NaN   
2                         AV ENTRE RIOS 2034       AVENIDA  ...  2034.0   
3      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...     NaN   
4  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...     NaN   

                                 Cruce  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                  DE LOS CORRALES AV.   
2                                  NaN   
3             VILLEGAS, CONRADO, GRAL.   
4              SAENZ PE?A, LUIS, PRES.   

                               Dirección Normalizada COMUNA  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                ENTRE RIOS AV. 2034      1   
3          LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4            SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   

                                 XY (CABA)         pos x         pos y  \
0    Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1    Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2   Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3    Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4  Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   

    PARTICIPANTES VICTIMA    ACUSADO  
0       MOTO-AUTO    MOTO       AUTO  
1  AUTO-PASAJEROS    AUTO  PASAJEROS  
2       MOTO-AUTO    MOTO       AUTO  
3         MOTO-SD    MOTO         SD  
4  MOTO-PASAJEROS    MOTO  PASAJEROS  

[5 rows x 21 columns]

In [18]:
hechos.columns

Index(['ID', 'N_VICTIMAS', 'FECHA', 'AAAA', 'MM', 'DD', 'HORA', 'HH',
       'LUGAR_DEL_HECHO', 'TIPO_DE_CALLE', 'Calle', 'Altura', 'Cruce',
       'Dirección Normalizada', 'COMUNA', 'XY (CABA)', 'pos x', 'pos y',
       'PARTICIPANTES', 'VICTIMA', 'ACUSADO'],
      dtype='object')

#### lo primero seria obtener las columnas a utilizar para el analisis

In [19]:
hechos.drop(columns=['HH', 'LUGAR_DEL_HECHO', 'Calle', 'Altura', 'Cruce', 'Dirección Normalizada', 'XY (CABA)', 'PARTICIPANTES'], inplace=True)

por que descartamos las columnas:

    -'HH': crearemos una nueva franja horaria acorde al analisis.
    -'LUGAR_DEL_HECHO', 'Calle', 'Altura', 'Cruce', 'Dirección Normalizada' : solo nos importa la geolocalizacion para graficar y analizar.
    -'XY (CABA)', 'PARTICIPANTES': es el dato anidado de otras columnas, por lo que es redundante.

In [20]:
hechos.head(5)

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA TIPO_DE_CALLE  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00       AVENIDA   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00      GRAL PAZ   
2  2016-0003           1 2016-01-03  2016   1   3  07:00:00       AVENIDA   
3  2016-0004           1 2016-01-10  2016   1  10  00:00:00       AVENIDA   
4  2016-0005           1 2016-01-21  2016   1  21  05:20:00       AVENIDA   

   COMUNA         pos x         pos y VICTIMA    ACUSADO  
0       8  -58.47533969  -34.68757022    MOTO       AUTO  
1       9  -58.50877521  -34.66977709    AUTO  PASAJEROS  
2       1  -58.39040293  -34.63189362    MOTO       AUTO  
3       8  -58.46503904  -34.68092974    MOTO         SD  
4       1  -58.38718297  -34.62246630    MOTO  PASAJEROS

In [21]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             696 non-null    object        
 1   N_VICTIMAS     696 non-null    int64         
 2   FECHA          696 non-null    datetime64[ns]
 3   AAAA           696 non-null    int64         
 4   MM             696 non-null    int64         
 5   DD             696 non-null    int64         
 6   HORA           696 non-null    object        
 7   TIPO_DE_CALLE  696 non-null    object        
 8   COMUNA         696 non-null    int64         
 9   pos x          696 non-null    object        
 10  pos y          696 non-null    object        
 11  VICTIMA        696 non-null    object        
 12  ACUSADO        696 non-null    object        
dtypes: datetime64[ns](1), int64(5), object(7)
memory usage: 70.8+ KB


#### verificamos los valores nulos para ello primero reemplazamos 'SD' por nulos

In [22]:
hechos.replace('SD', pd.NA, inplace=True)

In [23]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             696 non-null    object        
 1   N_VICTIMAS     696 non-null    int64         
 2   FECHA          696 non-null    datetime64[ns]
 3   AAAA           696 non-null    int64         
 4   MM             696 non-null    int64         
 5   DD             696 non-null    int64         
 6   HORA           695 non-null    object        
 7   TIPO_DE_CALLE  696 non-null    object        
 8   COMUNA         696 non-null    int64         
 9   pos x          696 non-null    object        
 10  pos y          696 non-null    object        
 11  VICTIMA        687 non-null    object        
 12  ACUSADO        673 non-null    object        
dtypes: datetime64[ns](1), int64(5), object(7)
memory usage: 70.8+ KB


#### ahora si vemos los datos nulos
 N°   Column                 Non-Null Count  Dtype         
---  ------         --------------  -----               
 6   HORA           695 non-null    object  ct        
 11  VICTIMA        687 non-null    object        
 12  ACUSADO        673 non-null    object 73 non-null    object null    object        

In [24]:
hechos[hechos['HORA'].isna()]

ID  N_VICTIMAS      FECHA  AAAA  MM  DD  HORA TIPO_DE_CALLE  \
518  2019-0103           1 2019-12-18  2019  12  18  <NA>      GRAL PAZ   

     COMUNA         pos x         pos y VICTIMA ACUSADO  
518      11  -58.52169422  -34.59471640    MOTO    MOTO

eliminamos la fila nulo de la columna HORA

In [25]:
hechos=hechos[hechos['ID']!='2019-0103']

eliminamos la fila nulo de la columna VICTIMA

In [26]:
hechos=hechos[~hechos['VICTIMA'].isna()]

evaluamos la calidad resultante del data set

In [27]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 686 entries, 0 to 695
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             686 non-null    object        
 1   N_VICTIMAS     686 non-null    int64         
 2   FECHA          686 non-null    datetime64[ns]
 3   AAAA           686 non-null    int64         
 4   MM             686 non-null    int64         
 5   DD             686 non-null    int64         
 6   HORA           686 non-null    object        
 7   TIPO_DE_CALLE  686 non-null    object        
 8   COMUNA         686 non-null    int64         
 9   pos x          686 non-null    object        
 10  pos y          686 non-null    object        
 11  VICTIMA        686 non-null    object        
 12  ACUSADO        668 non-null    object        
dtypes: datetime64[ns](1), int64(5), object(7)
memory usage: 75.0+ KB


In [28]:
print(f'''
El 100% del dataset es de 696 elementos como minimo deberiamos quedarnos con al menos el 95% de los datos para asegurar la calidad.
El 95% seria {696*0.95:.2f}, actualmente tenemos {686}, que seria el {(686*100)/696:.2f}%, ahora aun tenemos valores nulos en ACUSADO.
Si tambien descartamos nulos en esta columna nos quedariamos con {668} datos, por lo que nos quedariamos con {(668*100)/696:.2f}%
''')


El 100% del dataset es de 696 elementos como minimo deberiamos quedarnos con al menos el 95% de los datos para asegurar la calidad.
El 95% seria 661.20, actualmente tenemos 686, que seria el 98.56%, ahora aun tenemos valores nulos en ACUSADO.
Si tambien descartamos nulos en esta columna nos quedariamos con 668 datos, por lo que nos quedariamos con 95.98%



procedemos a eliminar datos nulos de la columna ACUSADO

In [29]:
hechos[hechos['ACUSADO'].isna()]

ID  N_VICTIMAS      FECHA  AAAA  MM  DD                 HORA  \
3    2016-0004           1 2016-01-10  2016   1  10             00:00:00   
32   2016-0045           1 2016-04-11  2016   4  11             10:25:00   
38   2016-0052           1 2016-04-20  2016   4  20             20:00:00   
57   2016-0080           1 2016-06-18  2016   6  18             06:42:00   
83   2016-0110           1 2016-08-24  2016   8  24             20:21:00   
105  2016-0135           1 2016-10-24  2016  10  24             05:00:00   
112  2016-0143           1 2016-11-06  2016  11   6             04:41:00   
119  2016-0151           1 2016-11-18  2016  11  18             20:35:00   
126  2016-0159           1 2016-12-04  2016  12   4             07:21:00   
160  2017-0025           1 2017-02-24  2017   2  24             17:05:00   
197  2017-0069           1 2017-05-24  2017   5  24             14:15:00   
216  2017-0091           1 2017-07-15  2017   7  15             13:00:00   
393  2018-0119           1 2018-11-05  2018  11   5             22:35:00   
397  2018-0123           1 2018-11-12  2018  11  12             03:32:00   
439  2019-0022           1 2019-03-15  2019   3  15  1900-01-02 00:00:00   
450  2019-0033           1 2019-04-18  2019   4  18             01:00:00   
457  2019-0040           1 2019-05-09  2019   5   9             12:45:00   
620  2021-0022           1 2021-02-28  2021   2  28             12:00:00   

    TIPO_DE_CALLE  COMUNA         pos x         pos y    VICTIMA ACUSADO  
3         AVENIDA       8  -58.46503904  -34.68092974       MOTO    <NA>  
32       GRAL PAZ      13  -58.46743471  -34.53476874       MOTO    <NA>  
38      AUTOPISTA      13             .             .       MOTO    <NA>  
57       GRAL PAZ       9  -58.52927982  -34.63931752       AUTO    <NA>  
83        AVENIDA       9  -58.51232237  -34.65903530     PEATON    <NA>  
105       AVENIDA      14  -58.42260225  -34.56403694     PEATON    <NA>  
112       AVENIDA       8  -58.44451316  -34.68475866       MOTO    <NA>  
119         CALLE       0             .             .     PEATON    <NA>  
126       AVENIDA       8  -58.44451316  -34.68475866       AUTO    <NA>  
160       AVENIDA       7  -58.43614426  -34.65424640       MOTO    <NA>  
197      GRAL PAZ       9  -58.52173071  -34.65997463       MOTO    <NA>  
216      GRAL PAZ      12  -58.48671535  -34.54321216       MOTO    <NA>  
393       AVENIDA       8  -58.43613623  -34.66292532     PEATON    <NA>  
397      GRAL PAZ       9  -58.50251790  -34.67450313       AUTO    <NA>  
439       AVENIDA       1  -58.39123782  -34.62352245       AUTO    <NA>  
450     AUTOPISTA       1  -58.36607137  -34.62150748       MOTO    <NA>  
457         CALLE       7  -58.44263580  -34.63679732       MOTO    <NA>  
620       AVENIDA      13  -58.45157681  -34.55217303  PASAJEROS    <NA>

en este proceso nos dimos cuenta que no tenemos la ubicacion gps de ciertos datos por lo que luego de este paso procederemos a tratarlo

In [30]:
hechos=hechos[~hechos['ACUSADO'].isna()]

In [31]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 668 entries, 0 to 695
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             668 non-null    object        
 1   N_VICTIMAS     668 non-null    int64         
 2   FECHA          668 non-null    datetime64[ns]
 3   AAAA           668 non-null    int64         
 4   MM             668 non-null    int64         
 5   DD             668 non-null    int64         
 6   HORA           668 non-null    object        
 7   TIPO_DE_CALLE  668 non-null    object        
 8   COMUNA         668 non-null    int64         
 9   pos x          668 non-null    object        
 10  pos y          668 non-null    object        
 11  VICTIMA        668 non-null    object        
 12  ACUSADO        668 non-null    object        
dtypes: datetime64[ns](1), int64(5), object(7)
memory usage: 73.1+ KB


eliminado los nulos de Acusado continuamos verificando la ubicacion gps

In [32]:
hechos[hechos['pos x']=='.']

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA TIPO_DE_CALLE  \
106  2016-0136           1 2016-10-25  2016  10  25  00:00:00     AUTOPISTA   
176  2017-0042           1 2017-04-10  2017   4  10  09:00:00      GRAL PAZ   
180  2017-0050           2 2017-04-28  2017   4  28  11:08:08     AUTOPISTA   
181  2017-0051           1 2017-05-01  2017   5   1  03:47:47     AUTOPISTA   
256  2017-0140           1 2017-11-19  2017  11  19  23:22:17     AUTOPISTA   
313  2018-0039           1 2018-04-21  2018   4  21  22:15:00     AUTOPISTA   
546  2020-0026           1 2020-05-17  2020   5  17  06:40:00     AUTOPISTA   
559  2020-0039           1 2020-09-01  2020   9   1  19:17:42         CALLE   
621  2021-0023           1 2021-03-01  2021   3   1  09:20:00     AUTOPISTA   

     COMUNA pos x pos y VICTIMA      ACUSADO  
106       4     .     .    MOTO       CARGAS  
176      14     .     .    MOTO       CARGAS  
180       9     .     .    MOTO       CARGAS  
181       7     .     .    AUTO         AUTO  
256       4     .     .    MOTO    PASAJEROS  
313      14     .     .  PEATON         AUTO  
546      14     .     .    MOTO  OBJETO FIJO  
559       9     .     .  PEATON       CARGAS  
621       4     .     .    MOTO       CARGAS

In [33]:
hechos[hechos['pos x']=='.'].count()

ID               9
N_VICTIMAS       9
FECHA            9
AAAA             9
MM               9
DD               9
HORA             9
TIPO_DE_CALLE    9
COMUNA           9
pos x            9
pos y            9
VICTIMA          9
ACUSADO          9
dtype: int64

In [34]:
round(((668-9)*100)/696,2)

94.68

si procedemos a eliminar los datos nulos de la ubicación quedariamos fuera del rango de datos de calidad por lo que procedemos a llenar segun la moda de la comuna y tipo de calle de cada uno

In [35]:
hechos.reset_index(inplace=True)
hechos.drop(columns=['index'], inplace=True)

In [36]:
hechos.iloc[1]['COMUNA']

9

In [37]:
hechos[hechos['COMUNA']==9]['pos x'].mode()[0]

'-58.50877521'

In [38]:
DF={'COMUNA':[4,7,9,14],'pos x':['.','.','.','.'],'pos y':['.','.','.','.'],'TIPO_DE_CALLE':['AUTOPISTA','AUTOPISTA','CALLE','AUTOPISTA']}
DF=pd.DataFrame(DF)
DF

COMUNA pos x pos y TIPO_DE_CALLE
0       4     .     .     AUTOPISTA
1       7     .     .     AUTOPISTA
2       9     .     .         CALLE
3      14     .     .     AUTOPISTA

In [39]:
hechos[(hechos['pos x']=='.') | (hechos['pos y']=='.')]

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA TIPO_DE_CALLE  \
96   2016-0136           1 2016-10-25  2016  10  25  00:00:00     AUTOPISTA   
160  2017-0042           1 2017-04-10  2017   4  10  09:00:00      GRAL PAZ   
164  2017-0050           2 2017-04-28  2017   4  28  11:08:08     AUTOPISTA   
165  2017-0051           1 2017-05-01  2017   5   1  03:47:47     AUTOPISTA   
236  2017-0140           1 2017-11-19  2017  11  19  23:22:17     AUTOPISTA   
292  2018-0039           1 2018-04-21  2018   4  21  22:15:00     AUTOPISTA   
519  2020-0026           1 2020-05-17  2020   5  17  06:40:00     AUTOPISTA   
532  2020-0039           1 2020-09-01  2020   9   1  19:17:42         CALLE   
593  2021-0023           1 2021-03-01  2021   3   1  09:20:00     AUTOPISTA   

     COMUNA pos x pos y VICTIMA      ACUSADO  
96        4     .     .    MOTO       CARGAS  
160      14     .     .    MOTO       CARGAS  
164       9     .     .    MOTO       CARGAS  
165       7     .     .    AUTO         AUTO  
236       4     .     .    MOTO    PASAJEROS  
292      14     .     .  PEATON         AUTO  
519      14     .     .    MOTO  OBJETO FIJO  
532       9     .     .  PEATON       CARGAS  
593       4     .     .    MOTO       CARGAS

In [40]:
i=hechos[(hechos['pos x']=='.') | (hechos['pos y']=='.')].index.to_list()

In [41]:
i

[96, 160, 164, 165, 236, 292, 519, 532, 593]

In [42]:
hechos.iloc[i]

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA TIPO_DE_CALLE  \
96   2016-0136           1 2016-10-25  2016  10  25  00:00:00     AUTOPISTA   
160  2017-0042           1 2017-04-10  2017   4  10  09:00:00      GRAL PAZ   
164  2017-0050           2 2017-04-28  2017   4  28  11:08:08     AUTOPISTA   
165  2017-0051           1 2017-05-01  2017   5   1  03:47:47     AUTOPISTA   
236  2017-0140           1 2017-11-19  2017  11  19  23:22:17     AUTOPISTA   
292  2018-0039           1 2018-04-21  2018   4  21  22:15:00     AUTOPISTA   
519  2020-0026           1 2020-05-17  2020   5  17  06:40:00     AUTOPISTA   
532  2020-0039           1 2020-09-01  2020   9   1  19:17:42         CALLE   
593  2021-0023           1 2021-03-01  2021   3   1  09:20:00     AUTOPISTA   

     COMUNA pos x pos y VICTIMA      ACUSADO  
96        4     .     .    MOTO       CARGAS  
160      14     .     .    MOTO       CARGAS  
164       9     .     .    MOTO       CARGAS  
165       7     .     .    AUTO         AUTO  
236       4     .     .    MOTO    PASAJEROS  
292      14     .     .  PEATON         AUTO  
519      14     .     .    MOTO  OBJETO FIJO  
532       9     .     .  PEATON       CARGAS  
593       4     .     .    MOTO       CARGAS

In [43]:
hechos.iloc[i][['pos x', 'pos y']]

pos x pos y
96      .     .
160     .     .
164     .     .
165     .     .
236     .     .
292     .     .
519     .     .
532     .     .
593     .     .

creamos una funcion para rellenar los valores del gps con la moda o con la mediana, de no encontrarse ninguno se queda nulo

In [44]:
def fill_gps(row):
    comuna = row['COMUNA']
    calle = row['TIPO_DE_CALLE']
    df = hechos[(hechos['COMUNA']==comuna) & (hechos['TIPO_DE_CALLE']==calle) & (hechos['pos x']!='.')]
    if df['pos x'].empty or df['pos y'].empty:
        return pd.Series({'pos x': pd.NA, 'pos y': pd.NA})
    modax = df['pos x'].mode()
    if modax.empty:
        modax = df['pos x']
        modax = modax.sort_values()
        if len(modax)%2==0:
            modax = modax[len(modax)//2]
        else:
            modax = modax[len(modax)//2+1]
    else:
        modax = modax[0]
    moday = df['pos y'].mode()
    if moday.empty:
        moday = df['pos y']
        moday = moday.sort_values()
        if len(moday)%2==0:
            moday = moday[len(moday)//2]
        else:
            moday = moday[len(moday)//2+1]
    else:
        moday = moday[0]
    return pd.Series({'pos x': modax, 'pos y': moday})

In [45]:
DF.apply(fill_gps,axis=1)

pos x         pos y
0  -58.37157668  -34.64035082
1  -58.43052224  -34.62941648
2  -58.48457413  -34.64143926
3  -58.40786274  -34.56726817

In [46]:
DF[['pos x','pos y']]=DF.apply(fill_gps, axis=1)

In [47]:
DF

COMUNA         pos x         pos y TIPO_DE_CALLE
0       4  -58.37157668  -34.64035082     AUTOPISTA
1       7  -58.43052224  -34.62941648     AUTOPISTA
2       9  -58.48457413  -34.64143926         CALLE
3      14  -58.40786274  -34.56726817     AUTOPISTA

contimuamos con el DF real

In [48]:
hechos.loc[i].apply(fill_gps, axis=1)

pos x         pos y
96   -58.37157668  -34.64035082
160          <NA>          <NA>
164  -58.47297505  -34.63659934
165  -58.43052224  -34.62941648
236  -58.37157668  -34.64035082
292  -58.40786274  -34.56726817
519  -58.40786274  -34.56726817
532  -58.48457413  -34.64143926
593  -58.37157668  -34.64035082

In [49]:
hechos.loc[i,['pos x','pos y']]

pos x pos y
96      .     .
160     .     .
164     .     .
165     .     .
236     .     .
292     .     .
519     .     .
532     .     .
593     .     .

In [50]:
hechos.loc[i, ['pos x', 'pos y']] = hechos.loc[i].apply(fill_gps, axis=1)

In [51]:
hechos.loc[i]

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA TIPO_DE_CALLE  \
96   2016-0136           1 2016-10-25  2016  10  25  00:00:00     AUTOPISTA   
160  2017-0042           1 2017-04-10  2017   4  10  09:00:00      GRAL PAZ   
164  2017-0050           2 2017-04-28  2017   4  28  11:08:08     AUTOPISTA   
165  2017-0051           1 2017-05-01  2017   5   1  03:47:47     AUTOPISTA   
236  2017-0140           1 2017-11-19  2017  11  19  23:22:17     AUTOPISTA   
292  2018-0039           1 2018-04-21  2018   4  21  22:15:00     AUTOPISTA   
519  2020-0026           1 2020-05-17  2020   5  17  06:40:00     AUTOPISTA   
532  2020-0039           1 2020-09-01  2020   9   1  19:17:42         CALLE   
593  2021-0023           1 2021-03-01  2021   3   1  09:20:00     AUTOPISTA   

     COMUNA         pos x         pos y VICTIMA      ACUSADO  
96        4  -58.37157668  -34.64035082    MOTO       CARGAS  
160      14          <NA>          <NA>    MOTO       CARGAS  
164       9  -58.47297505  -34.63659934    MOTO       CARGAS  
165       7  -58.43052224  -34.62941648    AUTO         AUTO  
236       4  -58.37157668  -34.64035082    MOTO    PASAJEROS  
292      14  -58.40786274  -34.56726817  PEATON         AUTO  
519      14  -58.40786274  -34.56726817    MOTO  OBJETO FIJO  
532       9  -58.48457413  -34.64143926  PEATON       CARGAS  
593       4  -58.37157668  -34.64035082    MOTO       CARGAS

In [52]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 668 entries, 0 to 667
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             668 non-null    object        
 1   N_VICTIMAS     668 non-null    int64         
 2   FECHA          668 non-null    datetime64[ns]
 3   AAAA           668 non-null    int64         
 4   MM             668 non-null    int64         
 5   DD             668 non-null    int64         
 6   HORA           668 non-null    object        
 7   TIPO_DE_CALLE  668 non-null    object        
 8   COMUNA         668 non-null    int64         
 9   pos x          667 non-null    object        
 10  pos y          667 non-null    object        
 11  VICTIMA        668 non-null    object        
 12  ACUSADO        668 non-null    object        
dtypes: datetime64[ns](1), int64(5), object(7)
memory usage: 68.0+ KB


In [53]:
round(((667)*100)/696,2)

95.83

eliminamos el dato que no pudimos rellenar la ubicacion

In [54]:
hechos = hechos[~((hechos['pos x'].isna()) | (hechos['pos y'].isna()))]

In [55]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 667 entries, 0 to 667
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             667 non-null    object        
 1   N_VICTIMAS     667 non-null    int64         
 2   FECHA          667 non-null    datetime64[ns]
 3   AAAA           667 non-null    int64         
 4   MM             667 non-null    int64         
 5   DD             667 non-null    int64         
 6   HORA           667 non-null    object        
 7   TIPO_DE_CALLE  667 non-null    object        
 8   COMUNA         667 non-null    int64         
 9   pos x          667 non-null    object        
 10  pos y          667 non-null    object        
 11  VICTIMA        667 non-null    object        
 12  ACUSADO        667 non-null    object        
dtypes: datetime64[ns](1), int64(5), object(7)
memory usage: 73.0+ KB


#### ahora con el Dataset libre de datos nulos procedemos a corregir el tipo de dato de cada columna.

In [56]:
hechos.loc[0,'ID']

'2016-0001'

In [57]:
hechos['ID'].str.replace('-', '')

0      20160001
1      20160002
2      20160003
3      20160005
4      20160008
         ...   
663    20210093
664    20210094
665    20210095
666    20210096
667    20210097
Name: ID, Length: 667, dtype: object

In [58]:
hechos['ID'] = hechos['ID'].str.replace('-', '')

In [59]:
hechos['ID'] = hechos['ID'].astype(int)
hechos['N_VICTIMAS'] = hechos['N_VICTIMAS'].astype(int)
hechos['AAAA'] = hechos['AAAA'].astype(int)
hechos['MM'] = hechos['MM'].astype(np.int8)
hechos['DD'] = hechos['DD'].astype(np.int8)
hechos['COMUNA'] = hechos['COMUNA'].astype(np.int8)

In [60]:
hechos.head(5)

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA TIPO_DE_CALLE  \
0  20160001           1 2016-01-01  2016   1   1  04:00:00       AVENIDA   
1  20160002           1 2016-01-02  2016   1   2  01:15:00      GRAL PAZ   
2  20160003           1 2016-01-03  2016   1   3  07:00:00       AVENIDA   
3  20160005           1 2016-01-21  2016   1  21  05:20:00       AVENIDA   
4  20160008           1 2016-01-24  2016   1  24  18:30:00       AVENIDA   

   COMUNA         pos x         pos y VICTIMA      ACUSADO  
0       8  -58.47533969  -34.68757022    MOTO         AUTO  
1       9  -58.50877521  -34.66977709    AUTO    PASAJEROS  
2       1  -58.39040293  -34.63189362    MOTO         AUTO  
3       1  -58.38718297  -34.62246630    MOTO    PASAJEROS  
4       8  -58.44451316  -34.68475866    MOTO  OBJETO FIJO

In [61]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 667 entries, 0 to 667
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             667 non-null    int32         
 1   N_VICTIMAS     667 non-null    int32         
 2   FECHA          667 non-null    datetime64[ns]
 3   AAAA           667 non-null    int32         
 4   MM             667 non-null    int8          
 5   DD             667 non-null    int8          
 6   HORA           667 non-null    object        
 7   TIPO_DE_CALLE  667 non-null    object        
 8   COMUNA         667 non-null    int8          
 9   pos x          667 non-null    object        
 10  pos y          667 non-null    object        
 11  VICTIMA        667 non-null    object        
 12  ACUSADO        667 non-null    object        
dtypes: datetime64[ns](1), int32(3), int8(3), object(6)
memory usage: 67.6+ KB


In [62]:
hechos[hechos['ID'].duplicated()]

Empty DataFrame
Columns: [ID, N_VICTIMAS, FECHA, AAAA, MM, DD, HORA, TIPO_DE_CALLE, COMUNA, pos x, pos y, VICTIMA, ACUSADO]
Index: []

no tenemos id duplicados veremos si existen filas duplicadas

In [63]:
hechos[hechos.duplicated()]

Empty DataFrame
Columns: [ID, N_VICTIMAS, FECHA, AAAA, MM, DD, HORA, TIPO_DE_CALLE, COMUNA, pos x, pos y, VICTIMA, ACUSADO]
Index: []

descartamos la existendia de datos duplicados

#### creamos una columna con la franja horaria: 'Madrugada', 'Mañana', 'Tarde', 'Noche'

In [64]:
hechos['HORA'] = pd.to_datetime(hechos['HORA'], format='%H:%M:%S')
bins = [0, 6, 12, 18, 24]
labels = ['Madrugada', 'Mañana', 'Tarde', 'Noche']
hechos['FH'] = pd.cut(hechos['HORA'].dt.hour, bins=bins, labels=labels, right=False)

In [65]:
hechos.head()

ID  N_VICTIMAS      FECHA  AAAA  MM  DD                HORA  \
0  20160001           1 2016-01-01  2016   1   1 1900-01-01 04:00:00   
1  20160002           1 2016-01-02  2016   1   2 1900-01-01 01:15:00   
2  20160003           1 2016-01-03  2016   1   3 1900-01-01 07:00:00   
3  20160005           1 2016-01-21  2016   1  21 1900-01-01 05:20:00   
4  20160008           1 2016-01-24  2016   1  24 1900-01-01 18:30:00   

  TIPO_DE_CALLE  COMUNA         pos x         pos y VICTIMA      ACUSADO  \
0       AVENIDA       8  -58.47533969  -34.68757022    MOTO         AUTO   
1      GRAL PAZ       9  -58.50877521  -34.66977709    AUTO    PASAJEROS   
2       AVENIDA       1  -58.39040293  -34.63189362    MOTO         AUTO   
3       AVENIDA       1  -58.38718297  -34.62246630    MOTO    PASAJEROS   
4       AVENIDA       8  -58.44451316  -34.68475866    MOTO  OBJETO FIJO   

          FH  
0  Madrugada  
1  Madrugada  
2     Mañana  
3  Madrugada  
4      Noche

ya con la categorización horaria procedemos a descartar la columa de hora

In [66]:
hechos.drop(columns=['HORA'], inplace=True)

In [67]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 667 entries, 0 to 667
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             667 non-null    int32         
 1   N_VICTIMAS     667 non-null    int32         
 2   FECHA          667 non-null    datetime64[ns]
 3   AAAA           667 non-null    int32         
 4   MM             667 non-null    int8          
 5   DD             667 non-null    int8          
 6   TIPO_DE_CALLE  667 non-null    object        
 7   COMUNA         667 non-null    int8          
 8   pos x          667 non-null    object        
 9   pos y          667 non-null    object        
 10  VICTIMA        667 non-null    object        
 11  ACUSADO        667 non-null    object        
 12  FH             667 non-null    category      
dtypes: category(1), datetime64[ns](1), int32(3), int8(3), object(5)
memory usage: 63.3+ KB


faltaria el dato sobre la fatalidad del siniestro, para ello debemos limpiar el siguiente Dataset

#### Continuamos con el Dataset Victimas

In [68]:
victimas.columns

Index(['ID_hecho', 'FECHA', 'AAAA', 'MM', 'DD', 'ROL', 'VICTIMA', 'SEXO',
       'EDAD', 'FECHA_FALLECIMIENTO'],
      dtype='object')

In [69]:
victimas.drop(columns=['FECHA', 'AAAA', 'MM', 'DD'], inplace=True)

descartamos datos de la fecha ya que en el dataset anterior ya la tenemos

In [70]:
victimas.head()

ID_hecho        ROL VICTIMA       SEXO EDAD  FECHA_FALLECIMIENTO
0  2016-0001  CONDUCTOR    MOTO  MASCULINO   19  2016-01-01 00:00:00
1  2016-0002  CONDUCTOR    AUTO  MASCULINO   70  2016-01-02 00:00:00
2  2016-0003  CONDUCTOR    MOTO  MASCULINO   30  2016-01-03 00:00:00
3  2016-0004  CONDUCTOR    MOTO  MASCULINO   18                   SD
4  2016-0005  CONDUCTOR    MOTO  MASCULINO   29  2016-02-01 00:00:00

#### Buscamos identificar datos nulos por lo que procedemos a reemplazar los 'SD?'

In [71]:
victimas.replace('SD', pd.NA, inplace=True)

In [72]:
victimas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717 entries, 0 to 716
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   ID_hecho             717 non-null    object
 1   ROL                  706 non-null    object
 2   VICTIMA              708 non-null    object
 3   SEXO                 711 non-null    object
 4   EDAD                 664 non-null    object
 5   FECHA_FALLECIMIENTO  649 non-null    object
dtypes: object(6)
memory usage: 33.7+ KB


en este caso no podremos eliminar datos de victimas ya que estan ligadas al dataset por lo que procedemos a extraer solo las victimas de quienes tenemos datos del siniestro en nuestro dataset

In [73]:
victimas.rename(columns={'ID_hecho':'ID'}, inplace=True)

In [74]:
victimas['ID'] = victimas['ID'].str.replace('-', '')

In [75]:
victimas['ID'] = victimas['ID'].astype(int)

In [76]:
hechos[['ID','FH']].head(5)

ID         FH
0  20160001  Madrugada
1  20160002  Madrugada
2  20160003     Mañana
3  20160005  Madrugada
4  20160008      Noche

In [77]:
dv = pd.merge(victimas, hechos[['ID','FH']], on = 'ID', how = 'inner')

In [78]:
dv

ID                   ROL VICTIMA       SEXO EDAD  \
0    20160001             CONDUCTOR    MOTO  MASCULINO   19   
1    20160002             CONDUCTOR    AUTO  MASCULINO   70   
2    20160003             CONDUCTOR    MOTO  MASCULINO   30   
3    20160005             CONDUCTOR    MOTO  MASCULINO   29   
4    20160008             CONDUCTOR    MOTO  MASCULINO   30   
..        ...                   ...     ...        ...  ...   
683  20210092                PEATON  PEATON   FEMENINO   50   
684  20210093  PASAJERO_ACOMPAÑANTE    MOTO   FEMENINO   18   
685  20210094  PASAJERO_ACOMPAÑANTE    MOTO   FEMENINO   43   
686  20210095             CONDUCTOR    MOTO  MASCULINO   27   
687  20210096             CONDUCTOR    AUTO  MASCULINO   60   

     FECHA_FALLECIMIENTO         FH  
0    2016-01-01 00:00:00  Madrugada  
1    2016-01-02 00:00:00  Madrugada  
2    2016-01-03 00:00:00     Mañana  
3    2016-02-01 00:00:00  Madrugada  
4    2016-01-24 00:00:00      Noche  
..                   ...        ...  
683  2021-12-12 00:00:00     Mañana  
684  2021-12-18 00:00:00      Tarde  
685  2021-12-20 00:00:00  Madrugada  
686  2022-01-02 00:00:00  Madrugada  
687  2021-12-20 00:00:00     Mañana  

[688 rows x 7 columns]

In [79]:
dv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688 entries, 0 to 687
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype   
---  ------               --------------  -----   
 0   ID                   688 non-null    int32   
 1   ROL                  686 non-null    object  
 2   VICTIMA              688 non-null    object  
 3   SEXO                 686 non-null    object  
 4   EDAD                 644 non-null    object  
 5   FECHA_FALLECIMIENTO  635 non-null    object  
 6   FH                   688 non-null    category
dtypes: category(1), int32(1), object(5)
memory usage: 30.6+ KB


por diversas circunstancias puede no tenerse el dato del ROL o del SEXO de la victima por lo que llenaremos con 'NO IDENTIFICADO'

In [80]:
dv['ROL'] = dv['ROL'].fillna('NO IDENTIFICADO')

In [81]:
dv['SEXO'] = dv['SEXO'].fillna('NO IDENTIFICADO')

In [82]:
dv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688 entries, 0 to 687
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype   
---  ------               --------------  -----   
 0   ID                   688 non-null    int32   
 1   ROL                  688 non-null    object  
 2   VICTIMA              688 non-null    object  
 3   SEXO                 688 non-null    object  
 4   EDAD                 644 non-null    object  
 5   FECHA_FALLECIMIENTO  635 non-null    object  
 6   FH                   688 non-null    category
dtypes: category(1), int32(1), object(5)
memory usage: 30.6+ KB


In [83]:
dv['ROL'] = dv['ROL'].astype('category')
dv['VICTIMA'] = dv['VICTIMA'].astype('category')
dv['SEXO'] = dv['SEXO'].astype('category')

In [84]:
dv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688 entries, 0 to 687
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype   
---  ------               --------------  -----   
 0   ID                   688 non-null    int32   
 1   ROL                  688 non-null    category
 2   VICTIMA              688 non-null    category
 3   SEXO                 688 non-null    category
 4   EDAD                 644 non-null    object  
 5   FECHA_FALLECIMIENTO  635 non-null    object  
 6   FH                   688 non-null    category
dtypes: category(4), int32(1), object(2)
memory usage: 17.1+ KB


llenamos la columna edad con la moda

In [85]:
dv['EDAD'] = dv['EDAD'].fillna(dv['EDAD'].mode()[0])

In [86]:
dv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688 entries, 0 to 687
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype   
---  ------               --------------  -----   
 0   ID                   688 non-null    int32   
 1   ROL                  688 non-null    category
 2   VICTIMA              688 non-null    category
 3   SEXO                 688 non-null    category
 4   EDAD                 688 non-null    int64   
 5   FECHA_FALLECIMIENTO  635 non-null    object  
 6   FH                   688 non-null    category
dtypes: category(4), int32(1), int64(1), object(1)
memory usage: 17.1+ KB


In [87]:
dv.drop(columns=['FH'], inplace=True)

con el df limpio vamos a separar el dataset con los datos de victimas fatales y sobrevivientes

#### extraemos los datos de los sobrevivientes donde la fecha de fallecimiento sea nulo, asi obtendremos sobrevivientes

In [88]:
victimas = dv[dv['FECHA_FALLECIMIENTO'].isna()].copy()

In [89]:
victimas = victimas.reset_index()

In [90]:
victimas.drop(columns=['FECHA_FALLECIMIENTO', 'index'], inplace=True)

In [91]:
victimas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53 entries, 0 to 52
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype   
---  ------   --------------  -----   
 0   ID       53 non-null     int32   
 1   ROL      53 non-null     category
 2   VICTIMA  53 non-null     category
 3   SEXO     53 non-null     category
 4   EDAD     53 non-null     int64   
dtypes: category(3), int32(1), int64(1)
memory usage: 1.6 KB


In [92]:
victimas['EDAD'] = victimas['EDAD'].astype(int)

In [93]:
victimas.to_parquet('../Datasets/victimas.parquet', index=False)

continuamos con el dataset de los fatales

In [94]:
dv = dv[~dv['FECHA_FALLECIMIENTO'].isna()].copy()

In [95]:
dv['EDAD'] = dv['EDAD'].astype(int)

In [96]:
dv.info()

<class 'pandas.core.frame.DataFrame'>
Index: 635 entries, 0 to 687
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype   
---  ------               --------------  -----   
 0   ID                   635 non-null    int32   
 1   ROL                  635 non-null    category
 2   VICTIMA              635 non-null    category
 3   SEXO                 635 non-null    category
 4   EDAD                 635 non-null    int32   
 5   FECHA_FALLECIMIENTO  635 non-null    object  
dtypes: category(3), int32(2), object(1)
memory usage: 17.4+ KB


In [97]:
dv['FECHA_FALLECIMIENTO'] = pd.to_datetime(dv['FECHA_FALLECIMIENTO'])

In [98]:
dv.info()

<class 'pandas.core.frame.DataFrame'>
Index: 635 entries, 0 to 687
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   ID                   635 non-null    int32         
 1   ROL                  635 non-null    category      
 2   VICTIMA              635 non-null    category      
 3   SEXO                 635 non-null    category      
 4   EDAD                 635 non-null    int32         
 5   FECHA_FALLECIMIENTO  635 non-null    datetime64[ns]
dtypes: category(3), datetime64[ns](1), int32(2)
memory usage: 17.4 KB


In [99]:
dv.head()

ID        ROL VICTIMA       SEXO  EDAD FECHA_FALLECIMIENTO
0  20160001  CONDUCTOR    MOTO  MASCULINO    19          2016-01-01
1  20160002  CONDUCTOR    AUTO  MASCULINO    70          2016-01-02
2  20160003  CONDUCTOR    MOTO  MASCULINO    30          2016-01-03
3  20160005  CONDUCTOR    MOTO  MASCULINO    29          2016-02-01
4  20160008  CONDUCTOR    MOTO  MASCULINO    30          2016-01-24

In [100]:
dv.to_parquet('../Datasets/victimasf.parquet', index=False)

ya tenemos los archivos para ambas victimas, ahora haremos la columna de fatalidad para hechos

In [106]:
victimasf = pd.read_parquet('../Datasets/victimasf.parquet')

In [107]:
hechos.head(5)

ID  N_VICTIMAS      FECHA  AAAA  MM  DD TIPO_DE_CALLE  COMUNA  \
0  20160001           1 2016-01-01  2016   1   1       AVENIDA       8   
1  20160002           1 2016-01-02  2016   1   2      GRAL PAZ       9   
2  20160003           1 2016-01-03  2016   1   3       AVENIDA       1   
3  20160005           1 2016-01-21  2016   1  21       AVENIDA       1   
4  20160008           1 2016-01-24  2016   1  24       AVENIDA       8   

          pos x         pos y VICTIMA      ACUSADO         FH  
0  -58.47533969  -34.68757022    MOTO         AUTO  Madrugada  
1  -58.50877521  -34.66977709    AUTO    PASAJEROS  Madrugada  
2  -58.39040293  -34.63189362    MOTO         AUTO     Mañana  
3  -58.38718297  -34.62246630    MOTO    PASAJEROS  Madrugada  
4  -58.44451316  -34.68475866    MOTO  OBJETO FIJO      Noche

In [108]:
victimasf.head()

ID        ROL VICTIMA       SEXO  EDAD FECHA_FALLECIMIENTO
0  20160001  CONDUCTOR    MOTO  MASCULINO    19          2016-01-01
1  20160002  CONDUCTOR    AUTO  MASCULINO    70          2016-01-02
2  20160003  CONDUCTOR    MOTO  MASCULINO    30          2016-01-03
3  20160005  CONDUCTOR    MOTO  MASCULINO    29          2016-02-01
4  20160008  CONDUCTOR    MOTO  MASCULINO    30          2016-01-24

In [109]:
victimas.head()

ID                   ROL VICTIMA       SEXO  EDAD
0  20160022  PASAJERO_ACOMPAÑANTE    MOTO  MASCULINO    41
1  20160027  PASAJERO_ACOMPAÑANTE    AUTO  MASCULINO    34
2  20160031             CONDUCTOR    MOTO  MASCULINO    21
3  20160048                PEATON  PEATON   FEMENINO    30
4  20160064                PEATON  PEATON  MASCULINO    20

In [110]:
victimasf[victimasf['ID']==20160022].count().iloc[0]

0

In [111]:
def fill_fatal (id):
    return victimasf[victimasf['ID']==id].count().iloc[0]

In [112]:
hechos['ID'].apply(fill_fatal)

0      1
1      1
2      1
3      1
4      1
      ..
663    1
664    1
665    1
666    1
667    1
Name: ID, Length: 667, dtype: int64

In [113]:
hechos['FATAL'] = hechos['ID'].apply(fill_fatal)

In [114]:
hechos.head()

ID  N_VICTIMAS      FECHA  AAAA  MM  DD TIPO_DE_CALLE  COMUNA  \
0  20160001           1 2016-01-01  2016   1   1       AVENIDA       8   
1  20160002           1 2016-01-02  2016   1   2      GRAL PAZ       9   
2  20160003           1 2016-01-03  2016   1   3       AVENIDA       1   
3  20160005           1 2016-01-21  2016   1  21       AVENIDA       1   
4  20160008           1 2016-01-24  2016   1  24       AVENIDA       8   

          pos x         pos y VICTIMA      ACUSADO         FH  FATAL  
0  -58.47533969  -34.68757022    MOTO         AUTO  Madrugada      1  
1  -58.50877521  -34.66977709    AUTO    PASAJEROS  Madrugada      1  
2  -58.39040293  -34.63189362    MOTO         AUTO     Mañana      1  
3  -58.38718297  -34.62246630    MOTO    PASAJEROS  Madrugada      1  
4  -58.44451316  -34.68475866    MOTO  OBJETO FIJO      Noche      1

In [115]:
hechos['FATAL'] = hechos['FATAL'].astype(np.int8)
hechos['N_VICTIMAS'] = hechos['N_VICTIMAS'].astype(np.int8)

In [116]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 667 entries, 0 to 667
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             667 non-null    int32         
 1   N_VICTIMAS     667 non-null    int8          
 2   FECHA          667 non-null    datetime64[ns]
 3   AAAA           667 non-null    int32         
 4   MM             667 non-null    int8          
 5   DD             667 non-null    int8          
 6   TIPO_DE_CALLE  667 non-null    object        
 7   COMUNA         667 non-null    int8          
 8   pos x          667 non-null    object        
 9   pos y          667 non-null    object        
 10  VICTIMA        667 non-null    object        
 11  ACUSADO        667 non-null    object        
 12  FH             667 non-null    category      
 13  FATAL          667 non-null    int8          
dtypes: category(1), datetime64[ns](1), int32(2), int8(5), object(5)
memory usage: 6

podemos optimizar el Dataset poniendo categoricos las columnas y en float la ubicacion

In [117]:
hechos['TIPO_DE_CALLE'] = hechos['TIPO_DE_CALLE'].astype('category')
hechos['VICTIMA'] = hechos['VICTIMA'].astype('category')
hechos['ACUSADO'] = hechos['ACUSADO'].astype('category')

In [132]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 667 entries, 0 to 667
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   ID             667 non-null    int32         
 1   N_VICTIMAS     667 non-null    int8          
 2   FECHA          667 non-null    datetime64[ns]
 3   AAAA           667 non-null    int32         
 4   MM             667 non-null    int8          
 5   DD             667 non-null    int8          
 6   TIPO_DE_CALLE  667 non-null    category      
 7   ID_COMUNA      667 non-null    int8          
 8   pos x          667 non-null    object        
 9   pos y          667 non-null    object        
 10  VICTIMA        667 non-null    category      
 11  ACUSADO        667 non-null    category      
 12  FH             667 non-null    category      
 13  FATAL          667 non-null    int8          
dtypes: category(4), datetime64[ns](1), int32(2), int8(5), object(2)
memory usage: 4

In [131]:
hechos.rename(columns={'COMUNA':'ID_COMUNA'}, inplace=True)

In [133]:
hechos.to_parquet('../Datasets/hechos.parquet', index=False)

In [129]:
hechos.head()

ID  N_VICTIMAS      FECHA  AAAA  MM  DD TIPO_DE_CALLE  COMUNA  \
0  20160001           1 2016-01-01  2016   1   1       AVENIDA       8   
1  20160002           1 2016-01-02  2016   1   2      GRAL PAZ       9   
2  20160003           1 2016-01-03  2016   1   3       AVENIDA       1   
3  20160005           1 2016-01-21  2016   1  21       AVENIDA       1   
4  20160008           1 2016-01-24  2016   1  24       AVENIDA       8   

          pos x         pos y VICTIMA      ACUSADO         FH  FATAL  
0  -58.47533969  -34.68757022    MOTO         AUTO  Madrugada      1  
1  -58.50877521  -34.66977709    AUTO    PASAJEROS  Madrugada      1  
2  -58.39040293  -34.63189362    MOTO         AUTO     Mañana      1  
3  -58.38718297  -34.62246630    MOTO    PASAJEROS  Madrugada      1  
4  -58.44451316  -34.68475866    MOTO  OBJETO FIJO      Noche      1

se ha obtenido el archivo c2022_caba_est_c2_1.xlsx, de la web oficial del INDEC sobre el censo del año 2022:

https://censo.gob.ar/index.php/datos_definitivos_caba/

https://censo.gob.ar/wp-content/uploads/2023/11/c2022_caba_est_c2_1.xlsx

de dicho archivo he extraido el dato poblacional y terrirotial de las comunas creando una nueva hoja poblacion,
ademas he hecho una tabla con los nombre de las comunas

In [119]:
poblacion = pd.read_excel('../Datasets/c2022_caba_est_c2_1.xlsx', sheet_name='poblacion')
comunas = pd.read_excel('../Datasets/c2022_caba_est_c2_1.xlsx', sheet_name='comuna')

In [120]:
poblacion

ID_COMUNA  SUPERFICIE EN KM2  POBLACIÓN TOTAL  DENSIDAD HAB/KM2
0           1               17.9           223554           12489.1
1           2                6.3           161645           25657.9
2           3                6.4           196240           30662.5
3           4               22.7           229240           10098.7
4           5                6.7           194271           28995.7
5           6                6.9           203043           29426.5
6           7               12.4           215896           17411.0
7           8               22.5           204367            9083.0
8           9               16.6           169063           10184.5
9          10               12.6           173004           13730.5
10         11               14.1           204601           14510.7
11         12               15.7           236887           15088.3
12         13               15.0           264385           17625.7
13         14               15.9           248635           15637.4
14         15               14.3           196876           13767.6

In [121]:
poblacion.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   ID_COMUNA          15 non-null     int64  
 1   SUPERFICIE EN KM2  15 non-null     float64
 2   POBLACIÓN TOTAL    15 non-null     int64  
 3   DENSIDAD HAB/KM2   15 non-null     float64
dtypes: float64(2), int64(2)
memory usage: 608.0 bytes


In [122]:
poblacion['ID_COMUNA'] = poblacion['ID_COMUNA'].astype(int)

In [123]:
poblacion.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   ID_COMUNA          15 non-null     int32  
 1   SUPERFICIE EN KM2  15 non-null     float64
 2   POBLACIÓN TOTAL    15 non-null     int64  
 3   DENSIDAD HAB/KM2   15 non-null     float64
dtypes: float64(2), int32(1), int64(1)
memory usage: 548.0 bytes


In [124]:
poblacion.to_parquet('../Datasets/poblacion.parquet', index=False)

In [125]:
comunas

ID_COMUNA                                             COMUNA
0           1  Retiro, San Nicolás, Puerto Madero, San Telmo,...
1           2                                           Recoleta
2           3                          Balvanera y San Cristóbal
3           4  La Boca, Barracas, Parque Patricios y Nueva Po...
4           5                                    Almagro y Boedo
5           6                                          Caballito
6           7                         Flores y Parque Chacabuco.
7           8      Villa Soldati, Villa Riachuelo y Villa Lugano
8           9             Liniers, Mataderos y Parque Avellaneda
9          10  Villa Real, Monte Castro, Versalles, Floresta,...
10         11  Villa General Mitre, Villa Devoto, Villa del P...
11         12  Coghlan, Saavedra, Villa Urquiza y Villa Pueyr...
12         13                       Núñez, Belgrano y Colegiales
13         14                                            Palermo
14         15  Chacarita, Villa Crespo, La Paternal, Villa Or...

In [126]:
comunas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   ID_COMUNA  15 non-null     int64 
 1   COMUNA     15 non-null     object
dtypes: int64(1), object(1)
memory usage: 368.0+ bytes


In [127]:
comunas['ID_COMUNA'] = comunas['ID_COMUNA'].astype(int)

In [128]:
comunas.to_parquet('../Datasets/comunas.parquet', index=False)